### PR TITLE
feat: add types for react-phone-number-input 3.1.5

### PR DESCRIPTION
**Pull request by opii972**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/react-phone-number-input/index.d.ts [hunk 1]
```diff
@@ -68,11 +75,15 @@ export interface CountrySelectComponentProps {
...
-    tabIndex?: number | string;
+    tabIndex?: number;
     /**
      * The currently selected country code
      */
     value?: string;
+    /**
+     * Language translations
+     */
+    labels?: { [key: string]: string };
```
Comment (strdr4605) [inline types/react-phone-number-input/index.d.ts:86]: From what I understand [labels](https://developer.mozilla.org/en-US/docs/Web/API/HTMLSelectElement/labels) is a read-only prop from a `<select/>` element. Is not a prop that can be passed!! 
Also, I looked in the [react-phone-number-input](https://gitlab.com/catamphetamine/react-phone-number-input) and did not see any usage in the `countrySelectComponent`.
What am I missing?

**Review by strdr4605 — CHANGES_REQUESTED** [review]

File: types/react-phone-number-input/index.d.ts (hunk 1 above)
Comment (opii972) [reply types/react-phone-number-input/index.d.ts:86]: In according to the documentation it is possible to pass a `labels` prop when you create a custom country `<select/>` https://gitlab.com/catamphetamine/react-phone-number-input/-/blob/master/README.md#L187.

Comment (strdr4605) [reply types/react-phone-number-input/index.d.ts:86]: I am not sure that this example is the same as [countrySelectComponent](https://gitlab.com/catamphetamine/react-phone-number-input/-/tree/master#countryselectcomponent) passed to `<PhoneInput/>`.

Comment (strdr4605) [reply types/react-phone-number-input/index.d.ts:86]: I don't use this package right now and don't sure if the `labels` prop is needed?
Did you have such a use case and typing was missing?
Maybe another interface should be used in this case?

Comment (opii972) [reply types/react-phone-number-input/index.d.ts:86]: Here is a more explicit example. You can do something like that:

```typescript
import fr from 'react-phone-number-input/locale/fr.json';

const CountrySelect = ({
   value,
   onChange,
   labels,
   ...rest
}: CountrySelectComponentProps) => (
    <select
        {...rest}
        value={value}
        onChange={(event) => {
            if (onChange) {
                onChange(event.target.value || undefined);
            }
        }}
    >
        {getCountries().map((country: string) => (
            <option key={country} value={country}>
                {labels[country]} +{getCountryCallingCode(country)}
            </option>
        ))}
    </select>
);

CountrySelect.defaultProps = {
    labels: fr
};

<PhoneInput
        value={'+64271231234'}
        onChange={(value: string) => {
            console.log(value);
        }}
        countrySelectComponent={CountrySelect}
/>
```

With the `labels` props you can apply translations.

Comment (strdr4605) [reply types/react-phone-number-input/index.d.ts:86]: Good, I would suggest also adding `labels` prop in `types/react-phone-number-input/react-phone-number-input-tests.tsx` as you did with `getCountries`.

File: types/react-phone-number-input/react-phone-number-input-tests.tsx [hunk 2]
```diff
@@ -43,3 +49,41 @@ const test2 = (
         smartCaret={false}
     />
 );
+
+const CountrySelect = ({
+   value,
+   onChange,
```
Comment (strdr4605) [inline types/react-phone-number-input/react-phone-number-input-tests.tsx:55]:
```suggestion
   onChange,
   labels,
```

File: types/react-phone-number-input/react-phone-number-input-tests.tsx [hunk 3]
```diff
@@ -43,3 +49,41 @@ const test2 = (
...
+        value={value}
+        onChange={(event) => {
+            if (onChange) {
+                onChange(event.target.value || undefined);
+            }
+        }}
+    >
+        {getCountries().map((country: string) => (
+            <option key={country} value={country}>
+                {country} +{getCountryCallingCode(country)}
```
Comment (strdr4605) [inline types/react-phone-number-input/react-phone-number-input-tests.tsx:69]:
```suggestion
                {labels[country]} +{getCountryCallingCode(country)}
```

**Review by strdr4605 — CHANGES_REQUESTED** [review]

File: types/react-phone-number-input/index.d.ts (hunk 1 above)
Comment (opii972) [reply types/react-phone-number-input/index.d.ts:86]: Added :)

**Review by strdr4605 — APPROVED** [review]

**Comment by opii972** [comment]
Ready to merge

**Pull request merged**